### DISCUSSION TITLE
How to declare Initial Conditions

### DISCUSSION THREAD

**Discussion by BadboyIntheLoop** [JSBSim-Team/jsbsim#596] [Q&A]
**Hello All JSBsim members,**

This Post I wanna discuss about Initial Conditions (IC), how to declare them in script file and <event> tag effect happenings when running! If you know, please explain for me with filed tags about:

1. Environment: Atmosphere, Temperature, Winds (normal winds, turbulents, downburst, gust, ... ). What tag I need to declare with them.
2. FG function: FG_RAMP, FG_STEP ... How it run?
3. States like Position, orientation and velecity, angular velecity are OK! I think we all know!!!

I know it seem to be HUGE but we can make it sense!!!

**Comment by seanmcleod** [comment]
I'd suggest you first take a look at the [JSBSim Reference Manual](https://jsbsim.sourceforge.net/JSBSimReferenceManual.pdf)

**Reply by agodemar** [reply]
Or else have a look at this [online reference manual](https://jsbsim-team.github.io/jsbsim-reference-manual/).

**Reply by BadboyIntheLoop** [reply]
Yep, i've read manual as soon as I approach JSBsim but it's not enough to me to declare more conditions. For instance, I know JSBsim can simulate the winds like normal wind, guts and turbulent... but I dont know what the xml tag would be used. I wanna discuss that in here!!!

**Reply by BadboyIntheLoop** [reply]
https://github.com/JSBSim-Team/jsbsim/blob/6461c864ca2570f8e0c63e6522bb911473d86e71/examples/FlightGear.cxx#L339-L350
Dear all, please help me show the xml file that has the tags like that!!! Thank you so much

**Comment by seanmcleod** [comment]
Example of defining a gust in the source code documentation.

https://github.com/JSBSim-Team/jsbsim/blob/375f5be0fdfeb210f4cf66f938e3e6fd426c2966/src/models/atmosphere/FGWinds.h#L107-L121

And the complete set of properties available for wind.

https://github.com/JSBSim-Team/jsbsim/blob/375f5be0fdfeb210f4cf66f938e3e6fd426c2966/src/models/atmosphere/FGWinds.cpp#L488-L564
